torch/_higher_order_ops/flex_attention.py

```diff
@@ -192,6 +192,14 @@ def math_attention(
     value = torch.repeat_interleave(value, G, dim=1)
     key = torch.repeat_interleave(key, G, dim=1)
 
+    Bq, Bkv = query.size(0), key.size(0)
+    if Bq != Bkv:
+        assert (
+            Bq > 1 and Bkv == 1
+        ), "Batch dimension must match. Otherwise, Bkv should be 1 and Bq should be larger than 1."
```

Comment:
Raise a runtime error, not an assert, and say `Bq and Bkv must broadcast. Got Bq={Bq} and Bkv={Bkv}`.